src/ScriptCs.Hosting/ScriptServicesBuilder.cs

```diff
@@ -48,7 +49,7 @@ public ScriptServices Build()
 
             _scriptEngineType = (Type)Overrides[typeof(IScriptEngine)];
 
-            var initDirectoryCatalog = _scriptName != null || _repl;
+            var initDirectoryCatalog = _scriptName != null || _repl || _loadScriptPacks;
```

Comment:
I have an idea using Nullable<bool>, this will allow me to know whether Load was set explicitly or not.